From a7d02b5b5e67c249012843b113c49b309172dffa Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Fri, 17 Mar 2017 11:44:31 +0800
Subject: [PATCH 1585/5242] MLK-14482-1: dma: imx-sdma: fix memory_2_memory
 copy failed

commit  bcd8dbc9bc0c6787784c792c4a51a1a2500ae794 from
https://source.codeaurora.org/external/imx/linux-imx.git

emi_2_emi have to be set in memory_2_memory case, code miss during
rebase, fix up it now.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/imx-sdma.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index e3534eb..2dcd252 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -875,6 +875,7 @@ static void sdma_get_pc(struct sdma_channel *sdmac,
 
 	switch (peripheral_type) {
 	case IMX_DMATYPE_MEMORY:
+		emi_2_emi = sdma->script_addrs->ap_2_ap_addr;
 		break;
 	case IMX_DMATYPE_DSP:
 		emi_2_per = sdma->script_addrs->bp_2_ap_addr;
-- 
1.7.9.5

